app/lib/repository.rb

```diff
@@ -456,8 +456,12 @@ def add(path, file_data=nil, mime_type=nil)
       @jobs.push(action: :add, path: path, file_data: file_data, mime_type: mime_type)
     end
 
-    def remove(path, expected_revision_identifier)
-      @jobs.push(action: :remove, path: path, expected_revision_identifier: expected_revision_identifier)
+    def remove(path, expected_revision_identifier, option)
```

Comment:
You've changed the signature of this method by adding a required argument called `option`. Have you made sure that every piece of code that calls `remove` will provide an argument for `option`? What might be a better way of passing this argument to the `remove` method?